javaslang/src/test/java/javaslang/collection/JavaConvertersTest.java

```diff
@@ -0,0 +1,178 @@
+        }
+
+        protected final java.util.List<Integer> javaList(Integer... ints) {
+            return java.util.Arrays.asList(ints);
+        }
+
+        // -- contains(Object)
+
+        @Test
+        public void shouldRecognizeThatEmptyListViewDoesNotContainElement() {
+            assertThat(empty().contains(1)).isFalse();
+        }
+
+        @Test
+        public void shouldRecognizeThatNonEmptyListViewDoesNotContainElement() {
+            assertThat(of(1).contains(2)).isFalse();
+        }
+
+        @Test
+        public void shouldRecognizeThatNonEmptyListViewContainElement() {
+            assertThat(of(1).contains(1)).isTrue();
+        }
+
+        @Test
+        public void shouldEnsureThatEmptyListViewContainsPermitsNullElements() {
+            assertThat(empty().contains(null)).isFalse();
+        }
+
+        @Test
+        public void shouldEnsureThatNonEmptyListViewContainsPermitsNullElements() {
+            assertThat(ofNull().contains(null)).isTrue();
+        }
+
+        @Test
+        public void shouldBehaveLikeStandardJavaWhenCallingContainsOfIncompatibleTypeOnEmptyListView() {
+            // DEV-NOTE: java.util.List#contains(Object) says:
+            //           @throws ClassCastException if the type of the specified element is incompatible with this list
+            //           but Java's List impls do not throw!
+            assertThat(empty().contains("")).isFalse();
+        }
+
+        @Test
+        public void shouldBehaveLikeStandardJavaWhenCallingContainsOfIncompatibleTypeOnNonEmptyListView() {
+            // DEV-NOTE: java.util.List#contains(Object) says:
+            //           @throws ClassCastException if the type of the specified element is incompatible with this list
+            //           but Java's List impls do not throw!
+            assertThat(of(1).contains("")).isFalse();
+        }
+
+        // TODO: more read-only tests
+
+    }
+
+    @RunWith(Parameterized.class)
+    public static class ImmutableListViewTest extends ListViewTest {
+
+        public ImmutableListViewTest(String name, ListFactory listFactory) {
+            super(listFactory);
+        }
+
+        // -- unit test parameterization
+
+        @Parameters(name = "{index}: {0}")
+        public static java.util.Collection<Object[]> data() {
+            // DEV-NOTE: It is sufficient to test
+            // - one standard Java impl (<-- ensures our tests are correct)
+            // - one LinearSeq impl
+            // - one IndexedSeq impl
+            // to ensure the correctness of the immutable ListView impl.
+            return java.util.Arrays.asList(new Object[][] {
+                    { java.util.List.class.getName(), ListFactory.of(java.util.Arrays::asList) },
+                    { List.class.getName(), ListFactory.of(ints -> List.of(ints).asJava()) },
+                    { Vector.class.getName(), ListFactory.of(ints -> Vector.of(ints).asJava()) }
```

Comment:
We do not need test all of our collections - we only need to test, if our view impl is correct for all general types that are supported (like IndexedSeq, LinearSeq, Set, SortedSet, ...)